examples/complete-example/main.tf

```diff
@@ -2,61 +2,96 @@
-    },
-    {
-      availability_zone = "us-east-1b",
-      cidr_block        = "10.0.96.0/21",
-    }
-  ]
+  # Define a mode on how to deploy NAT gateways for private networks.
+  # Valid values are:
+  # - "single" to create a single NAT gateway in the first availability zone to route all private traffic to
+  # - "one_per_az" to create a NAT gateway per availability zone that has a private subnet. This needs a public subnets in the same AZs.
+  nat_gateway_mode = "single"
 
-  private_subnets = [
-    {
-      availability_zone = "us-east-1a",
-      cidr_block        = "10.0.112.0/21",
-    },
+  subnets = [
     {
-      availability_zone = "us-east-1b",
-      cidr_block        = "10.0.128.0/21",
+      # define a group name for the subnets. This can be any string. Default is "main".
+      group = "main"
+
+      # Class of the subnet: Default is "private".
+      # - "public" defines a set of subnets where delployed components can be reachable via the public internet.
+      # - "private" defines a set subnets where components are not publicly reachable but can reach the internet.
+      # - "intra" (in development) defines a set of subnets that has not connectivity to the public internet.
+      class = "public"
+
+      # defines whether components deployed into the subnet will be assigned a public IPv4 address when launched.
+      map_public_ip_on_launch = true
+
+      # Define the base CIDR Block of the subnets and the parameters to calculate each CIDR Block. Default is the VPC CIDR Block.
+      #   This example: devide the VPC CIDR Block into 2^4 = 16 blocks and use the first block as a base for all subnets
+      #                 Add 4 bits to the prefix of "10.0.0.0/16" results in "10.0.0.0/20".
+      cidr_block = cidrsubnet(local.cidr_block, 4, 0)
+
+      # How many bits should be added when calculating the subnets CIDR Blocks. Default is 8.
+      #   This example: For each subnet add another 4 bits to the prefix, so each netnum will result in:
+      #                 netnum=0 => subnet_cidr_block="10.0.0.0/24"
+      #                 netnum=1 => subnet_cidr_block="10.0.1.0/24"
+      #                  ...
+      #                 netnum=15 => subnet_cidr_block="10.0.15.0/24"
+      newbits = 4
+
+      # A map of subnets keyed by availability zone suffix. The Number defined the network number with in the CIDR Block.
+      # see https://www.terraform.io/docs/configuration/functions/cidrsubnet.html for details on how this is calculated internally
+      #  Note: When adjusting cidr_block or newbits you might also need to adjust the netnums.
+      #   This Example: Deploy one subnet in availability zone "a" ("10.0.0.0/24") and one subnet in availability zone "b" ("10.0.1.0/24")
+      netnums_by_az = {
+        a = [0] # "10.0.0.0/24"
+        b = [1] # "10.0.1.0/24"
+      }
+
+      # subnets_by_az = {
+      #   a = ["10.0.0.0/24"]
+      #   b = ["10.0.1.0/24"]
+      # }
```

Comment:
@soerenmartius this could be an alternative format..  classic...doe the coments make sense?

we could also support both...

Reply:
I'm usually a fan of one consistent API that offers one way of achieving things - in that particular case it could make sense to offer both solutions though. We need to make sure to really offer consistent and comprehensive documentation and tests for this module. After all, I see this becoming one of our flagship modules which we should promote on reddit and hackernews once we have a stable version in place.